src/frontend/context/ObservationsContext.tsx

```diff
@@ -112,6 +115,15 @@ export const ObservationsProvider = ({
     };
   }, [state.reload]);
 
+  React.useEffect(() => {
+    if (authState === "obscured") {
+      dispatch({ type: "reload_success", value: [] });
+      return;
+    }
+
+    dispatch({ type: "reload" });
+  }, [authState]);
+
```

Comment:
I think rather than this approach, better to maybe modify the contextValue instead, to avoid the need to reloads and useEffect, which adds complexity to this already complicated context. E.g.

```js
  const contextValue: ObservationsContextType = React.useMemo(() => {
    const derivedState = { ...state }
    if (authState === 'obscured') derivedState.observations = {}
    return [derivedState, dispatch]
  },
     [state, authState]
   );